Promote hunspell over spell/aspell to avoid surprises where hunspell is preferred

--- Test-Spelling/lib/Test/Spelling.pm
+++ Test-Spelling/lib/Test/Spelling.pm
@@ -32,10 +32,10 @@ sub spellchecker_candidates {
     return $SPELLCHECKER if $SPELLCHECKER;
 
     return (
-        'spell', # for back-compat, this is the top candidate ...
-        'aspell list -l en', # ... but this should become first soon
-        'ispell -l',
         'hunspell -l',
+        'aspell list -l en',
+        'ispell -l',
+        'spell',
     );
 }
 
